workers/datasets_based/src/datasets_based/worker.py

```diff
@@ -296,6 +297,7 @@ def process(
             try:
                 self.pre_compute()
                 content = self.compute()
+                validate_content_size(content=content, content_max_size=self.common_config.content_max_size)
```

Comment:
move to the CachedResponse.upsert_response method?

Reply:
As per last comment, I kept it on the worker